### DISCUSSION TITLE
Database operations

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#524] [Q&A] [answered]
Hi,

When exporting the database, which file format would you recommand? SQL, CSV, TSV?
What about the output: open, save? What do they mean?

What are the best settings in the export view to save the database securely so I can later re-import the database in case there is a problem.

Thank you.

**Comment by matthew73210** [comment]
Hey,

Open just opens it in a new tab, then you can copy etc...
Save will download the file per which extension you chose. I'd go with .csv as you can also do some python fun on it.
tsv is just a varient of csv but allows full text by using a tab instead of , or ;
Prehabs the best option would be to export multiple varients in different file formats, that way you are safe.

**Reply by blogdemoi** [reply]
Thank you.

**Comment by mcguirepr89** [comment, marked as answer]
@blogdemoi Since the database is just a file, the BEST way to back it up is to make a copy of the file. The file is "`birds.db`" -- snag a copy periodically to keep on hand in case you need to restore things that way. You can use the "Tools" > "File Manager" to locate and download your copy of the database.

**Reply by blogdemoi** [reply]
You can use the "Tools" > "File Manager" to locate and download your copy of the database.

I can't find the location of birds.db using the File Manager.

If I go to the "Database Maintenance" I find this:

![Vtjw7dc](https://user-images.githubusercontent.com/83965802/187993215-27dde68b-a2ea-4177-a1ee-b9e0f6b171fd.png)

Then I select: 
-Output: sav
-Format: CSV,
-Tables: drop+create, auto increment (not selected), triggers (selected)
-Data: insert

Below I have selected: tables and data

Then I click on the export button and I save a detections.csv file. Is it the way to do the right file to save so I can import this file in case something goes wrong?

Thank you.

**Reply by ehpersonal38** [reply]
I wouldn't muck around with the Export command in Database Maintenance. Just copy `birds.db` somewhere safe and you'll be set.

`birds.db` is located at `/home/$USER/BirdNET-Pi/scripts/birds.db`. 

where `$USER` should be replaced with your Raspberry Pi user, e.g. `pi`.

**Reply by blogdemoi** [reply]
Thank you.
